microsoftbotframework/response.py

```diff
@@ -150,14 +152,14 @@ def _request(self, response_url, method, response_json=None):
         logger.info('response_headers: {}'.format(json.dumps(self.headers)))
         logger.info('response_json: {}'.format(json.dumps(response_json)))
 
-        post_response = request_method(response_url, json=response_json, headers=self.headers)
+        post_response = request_method(response_url, timeout=self.timeout_seconds, json=response_json, headers=self.headers)
 
         if 300 > post_response.status_code >= 200:
             logger.info('Successfully posted to Microsoft Bot Connector. {}'.format(post_response.text).replace('\n', '').replace('\r', ''))
         else:
             logger.error('Error posting to Microsoft Bot Connector. Status Code: {}, Text {}'
                          .format(post_response.status_code, post_response.text))
-
+        post_response.raise_for_status()
```

Comment:
Please invoke raise_for_status right after the request_method call.
post_response.status_code does not exist if a Connection Error or Timeout was raised.